core/src/main/java/com/amplifyframework/core/model/query/predicate/Evaluable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model.query.predicate;
+
+/**
+ * Interface for evaluable item.
+ * @param <T> Type of target value being evaluated against
+ */
+public interface Evaluable<T> {
```

Comment:
If this is only ever used by the `QueryOperator`s, you could consider baking the `evalute(T target)` method right into that interface.

Reply:
Also used by `QueryPredicate`